### ISSUE TITLE
openai whisper

### ISSUE THREAD

**Issue by SomeCodingUser** [Caldarie/flutter_tflite_audio#44]
Hi,

could you use this library to run openai whisper with a tflite model? In the examples there are always labels provided, but for whisper there would not be any labels I think?

Thanks!

**Comment by Caldarie** [comment]
Hi,

I had a quick look and it seems openai whisper uses pytorch. For it to be compatible with this plugin:

1. You’ll need to convert it to tflite. I believe it can be done if you follow the instructions [here](https://ai.aioz.io/guides/ml-ops/torch2tflite_convert/)
2. After converting the model, run the model through [here](https://netron.app/) to find input and output.
3. As for the labels.. you don’t need to have it. As long as the input can receive audio, and the output returns an array then it should work.
4. If input or output size does not match to this plugin, you may need to tinker the code to match your needs. Do let me know if you reach this stage 


Also be aware that tflite does have a lot of limitations. For more information, read about operator compatibilities [here](https://www.tensorflow.org/lite/guide/ops_compatibility). When converting from PyTorch to TensorFlow models, there may be some incompatibilities with your converted model on mobile devices

**Comment by nyadla-sys** [comment]
This is proof-of-concept project to create an Android app based on Whisper TFLite, which leverages the stock Android UI to show off its features.
[Whisper-TFLIte-Android-Example](https://github.com/usefulsensors/openai-whisper/tree/main/android_app/Whisper-TFLIte-Android-Example)

Refer here for Whidper TFLite model
https://github.com/usefulsensors/openai-whisper/blob/main/models/whisper.tflite

go thru README here 
https://github.com/usefulsensors/openai-whisper

**Comment by Caldarie** [comment]
@nyadla-sys Thanks for the share. :)

**Comment by nyadla-sys** [comment]
Feel free to download the openai/whisper-tiny tflite-based Android Whisper ASR APP from [Google App Store](https://play.google.com/store/apps/details?id=com.whisper.android.tflitecpp).
